src/transformers/tokenization_utils_base.py

```diff
@@ -843,22 +845,23 @@ def __init__(self, verbose=True, **kwargs):
                         isinstance(t, (str, AddedToken)) for t in value
                     ), "One of the tokens is not a string or an AddedToken"
                     setattr(self, key, value)
-                elif isinstance(value, (str, AddedToken)):
+                elif isinstance(value, (str)):
+                    value = AddedToken(value, normalized=False)
+                    setattr(self, key, value)
+                elif isinstance(value, AddedToken):
                     setattr(self, key, value)
                 else:
-                    raise TypeError(f"special token {key} has to be either str or AddedToken but got: {type(value)}")
+                    raise TypeError(f"Special token {key} has to be either str or AddedToken but got: {type(value)}")
 
     def sanitize_special_tokens(self) -> int:
         """
-        Make sure that all the special tokens attributes of the tokenizer (`tokenizer.mask_token`,
-        `tokenizer.cls_token`, etc.) are in the vocabulary.
-
-        Add the missing ones to the vocabulary if needed.
-
-        Return:
-            `int`: The number of tokens added in the vocabulary during the operation.
+        The `sanitize_special_tokens` is now deprecated and does not do anything. It is only kept for backward
```

Comment:
I'm not sure we should just deprecate the behaviour directly like this, this isn't backward compatible. This should really be check for two releases then removed.